Tests/DatadogTests/Datadog/RUM/RUMVitals/VitalRefreshRateReaderTests.swift

```diff
@@ -97,4 +97,133 @@ class VitalRefreshRateReaderTests: XCTestCase {
+        let thirdFps = reader.framesPerSecond(provider: frameInfoProvider)
+        XCTAssertEqual(thirdFps, 60)
+
+        // fourth frame recorded
+        frameInfoProvider.currentFrameTimestamp = 0.032
+        let fourthFps = reader.framesPerSecond(provider: frameInfoProvider)
+        XCTAssertEqual(fourthFps, 30)
+    }
+
+    /* Rate representation
+     *
+     * 0----------8ms----------------------------33ms---------43ms
+     * |    8ms    |             25ms             |    10ms    |
+    */
+    func testFramesPerSecond_givenAdaptiveSyncDisplay() {
+        let reader = VitalRefreshRateReader(notificationCenter: mockNotificationCenter)
+        var frameInfoProvider = FrameInfoProviderMock(maximumDeviceFramesPerSecond: 120)
+
+        // first frame recorded
+        frameInfoProvider.currentFrameTimestamp = 0
+        frameInfoProvider.nextFrameTimestamp = 0.008
+        let firstFps = reader.framesPerSecond(provider: frameInfoProvider)
+        XCTAssertNil(firstFps)
+
+        // second frame recorded
+        frameInfoProvider.currentFrameTimestamp = 0.008
+        frameInfoProvider.nextFrameTimestamp = 0.033
+        let secondFps = reader.framesPerSecond(provider: frameInfoProvider)
+        XCTAssertEqual(secondFps, 60)
+
+        // third frame recorded
+        frameInfoProvider.currentFrameTimestamp = 0.033
+        frameInfoProvider.nextFrameTimestamp = 0.043
+        let thirdFps = reader.framesPerSecond(provider: frameInfoProvider)
+        XCTAssertEqual(thirdFps, 60)
+
+        // fourth frame recorded
+        frameInfoProvider.currentFrameTimestamp = 0.043
+        let fourthFps = reader.framesPerSecond(provider: frameInfoProvider)
+        XCTAssertEqual(fourthFps, 60)
+    }
+
+    /* Rate representation
+     *
+     * 0----------8ms----------------------------33ms---------43ms
+     * |    8ms    |             25ms             |    10ms    |
+     *                                         skipped
+    */
+    func testFramesPerSecond_givenAdaptiveSyncDisplayWithFreezingFrames() {
+        let reader = VitalRefreshRateReader(notificationCenter: mockNotificationCenter)
+        var frameInfoProvider = FrameInfoProviderMock(maximumDeviceFramesPerSecond: 120)
+
+        // first frame recorded
+        frameInfoProvider.currentFrameTimestamp = 0
+        frameInfoProvider.nextFrameTimestamp = 0.008
+        let firstFps = reader.framesPerSecond(provider: frameInfoProvider)
+        XCTAssertNil(firstFps)
+
+        // second frame recorded
+        frameInfoProvider.currentFrameTimestamp = 0.008
+        frameInfoProvider.nextFrameTimestamp = 0.033
+        let secondFps = reader.framesPerSecond(provider: frameInfoProvider)
+        XCTAssertEqual(secondFps, 60)
+
+        // third frame recorded
+        frameInfoProvider.currentFrameTimestamp = 0.043
+        let thirdFps = reader.framesPerSecond(provider: frameInfoProvider)
+        XCTAssertEqual(thirdFps, 42.85714285714286)
```

Comment:
flaky ❄️/ This looks flaky - we [use `accuracy:`](https://github.com/search?q=repo%3ADataDog%2Fdd-sdk-ios+accuracy&type=code) to avoid random failures on CI (happened in the past):
```suggestion
        XCTAssertEqual(thirdFps, 42.8571, accuracy: 0.0001)
```

Reply:
we shouldn't get flakiness on this as this is just pure math unless Swift changes their floating point representation.

I general, I agree but here approximation is not even required.